opal/datatype/opal_copy_functions_heterogeneous.c

```diff
@@ -89,13 +89,14 @@ struct bit80 {
 static inline void
 opal_dt_swap_long_double(void *to_p, const void *from_p, const size_t size, size_t count, uint32_t remoteArch)
 {
+
 #ifdef HAVE_IEEE754_H
-    size_t i;
-    long double*to = (long double *) to_p;
 
     if ((opal_local_arch&OPAL_ARCH_LDISINTEL) && !(remoteArch&OPAL_ARCH_LDISINTEL)) {
 #ifdef __x86_64
-        for (i=0; i<count; i++, to++) {
+        long double*to = (long double *) to_p;
```

Comment:
To fix the CI compiler failure and make sure we preserve functionality, I moved the temporary pointer `to` back inside of each #define as a new fix.